include/zephyr/drivers/rtc.h

```diff
@@ -0,0 +1,454 @@
+
+	return api->set_time(dev, timeptr);
+}
+
+/**
+ * @brief API for getting RTC time
+ *
+ * @param dev Device instance
+ * @param timeptr Destination for the time
+ *
+ * @return 0 if successful
+ * @return -ENODATA if time has not been set
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_get_time(const struct device *dev, struct tm *timeptr);
+
+static inline int z_impl_rtc_get_time(const struct device *dev, struct tm *timeptr)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->get_time(dev, timeptr);
+}
+
+/**
+ * @brief API for getting the supported fields of the RTC alarm time
+ *
+ * @param dev Device instance
+ * @param mask Mask of fields in the alarm time which are supported
+ *
+ * @return 0 if successful
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_get_supported_fields(const struct device *dev, uint16_t *mask);
+
+static inline int z_impl_rtc_alarm_get_supported_fields(const struct device *dev, uint16_t *mask)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_get_supported_fields(dev, mask);
+}
+
+/**
+ * @brief API for setting RTC alarm time
+ *
+ * @details To enable an RTC alarm, one or more fields of the RTC alarm time
+ * must be enabled. The mask designates which fields of the RTC alarm time to
+ * enable. If the mask parameter is 0, the alarm will be disabled. The RTC
+ * alarm will trigger when all enabled fields of the alarm time match the RTC
+ * time.
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm
+ * @param mask Mask of fields in the alarm time to enable
+ * @param timeptr The alarm time to set
+ *
+ * @note The timeptr param may be NULL if the mask param is 0
+ *
+ * @note Only the enabled fields in the timeptr param need to be set
+ *
+ * @note Bits in the mask param are defined in the RTC_ALARM_TIME_MASK defines
+ *
+ * @return 0 if successful
+ * @return -EINVAL if id is out of range or time is invalid
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+				 const struct tm *timeptr);
+
+static inline int z_impl_rtc_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+					    const struct tm *timeptr)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_set_time(dev, id, mask, timeptr);
+}
+
+/**
+ * @brief API for getting RTC alarm time
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm
+ * @param mask Destination for mask of fields which are enabled in the alarm time
```

Comment:
Does this return the values provided when setting the alarm, or the values from when the alarm triggered?

If it's the former, then is it needed?

Reply:
These are the values that the alarm is configured with, if the alarm is triggered, the values here will reflect the value it matched with. I will expand the alarm_set_time description to note that the pending state must be cleared when alarm time is set.